File: frontend/src/services/summary.ts
// src/services/summary.ts
import api from "@/lib/axios";
import { MonthlySummaryResponse } from "@/models/summary";

export const summaryService = {
  async getMonthlySummary(year: number, month: number): Promise<MonthlySummaryResponse> {
    const { data } = await api.get<MonthlySummaryResponse>("/summary/monthly", {
      params: { year, month },
    });
    return data;
  },
};

File: backend/app/routers/summary.py
# app/routers/summary.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.dependencies.db import get_db
from app.schemas.summary import MonthlySummaryResponse
from app.services.summary import SummaryService

router = APIRouter(tags=["summary"])


@router.get("/summary/monthly", response_model=MonthlySummaryResponse)
def get_monthly_summary(
    year: int = Query(..., description="Year (e.g. 2026)"),
    month: int = Query(..., description="Month (1-12)"),
    db: Session = Depends(get_db),
):
    return SummaryService.calculate_monthly_summary(db, year, month)
